2021-01-Winter/0_modified_binary_search/README.md

```diff
@@ -99,57 +141,207 @@ Go to [Solution](#s3)   [Top](#top)
 
 ### 1. SOLUTION 1 TODO :bug:
 
-Source: TODO :bug:
-
-#### Naive/Simple Solution
-
-TODO :bug:
-
-#### Optimal Solution
-
-TODO :bug:
-
-#### Testing The Solutions OR Driver For Solution
-
-TODO :bug:
+Source: https://leetcode.com/problems/binary-search/
+
+#### Binary Search Solution
+
+1. First, lets initiate our `left` and `right` variables, our left will start at the 0th index of the array, and right will start at `arrays.length -1` (the very last index of the array.)
+2. Lets make a while loop, `while(left <= right)`
+3. Inside the while loop we'll initiate the pivot index or what some call the "mid," so `int pivot = (left + right) /2;`
+4. Create if statement `if(nums[pivot] == target])`, then we've found target, `return pivot`.
+5. Create else if statement, `else if(target < nums[pivot])`, then `right = pivot - 1;`
+6. Create else statement, `else`, `left = pivot + 1`.
+7. Outside of our while loop, we will `return -1`, when we reach here, we've searched through our whole array and failed to find our target index, so return -1.
+
+Complexity: O(N log N) since we divide the array in half each time.
+Space Complexity: o(1) no extra space used 
+
+```
+Example Code of Solution
+    public int search(int[] nums, int target) {
+        int left = 0, right = nums.length - 1; //Initialize two pointers, one at the beginning of the array, and one at the very end.
+        while(left <= right){
+            int pivot = (left + right) / 2; //Pivot starts at the middle.
+            if(nums[pivot] == target)
+                return pivot; //Our target is found, return index.
+            else if(target < nums[pivot])
+                right = pivot - 1; //If our target is less than our pivot, we close out the right side from mid, since we know our target isn't there.
+            else
+                left = pivot + 1; //Our target is greater than pivot, close out the left side from mid.
+        }
+        return -1; //Return -1 if not found.
+        }
```

Comment:
nit: move this closing bracket back a few spaces so it matches with the bracket on line 161